### PR TITLE
Add method in ContactSelf to update name and signature

### PR DESCRIPTION
**Pull request by windmemory**
This change is a prototype. It is adding two new method to `ContactSelf` class. 

* `setName` to change the nick name of the bot
* `signature` to change the signature of the bot

Will provide c9 playground link for working prototype in following comment

### CODE REVIEW

**Comment by windmemory** [comment]
Here is the [c9 playgound](https://ide.c9.io/lijiarui/update-self-info#openfile-README.md)
This is the [`wechaty-puppet` PR](https://github.com/Chatie/wechaty-puppet/pull/15)
This is the [`wechaty-puppet-padchat` PR](https://github.com/lijiarui/wechaty-puppet-padchat/pull/127)

Please let me know your thought.

File: src/user/contact-self.ts [hunk 1]
```diff
@@ -115,4 +115,21 @@ export class ContactSelf extends Contact {
     return qrcodeData
   }
 
+  public async setName (name: string): Promise<boolean> {
+    log.verbose('Contact-self', 'setName()')
```
Comment (zixia) [inline src/user/contact-self.ts]: 'Contact-self' -> 'ContactSelf'

**Comment by zixia** [comment]
Could you be able to use the latest version of c9 ide on amazon cloud? If you can, let's switch to the new Amazon cloud ide in the future with your own account for your convenience.

File: src/user/contact-self.ts (hunk 1 above)
Comment (windmemory) [reply src/user/contact-self.ts]: Will do the change.

**Comment by windmemory** [comment]
1. I would like to try the amazon c9, will try to use amazon c9 next time. 

2. I am also hate the long merge commits in the PR, but have no better idea of getting rid of that.. This is weird, I don't have these kind of commits on `wehcaty-puppet`'s PR. Will do this when I got time.

Any other comments besides the `Contact-self` one with this PR ?

**Comment by zixia** [comment]
Thank you very much for the new features!

**Pull request merged**